messages/teleporter/message_handler.go

```diff
@@ -167,7 +183,47 @@ func (m *messageHandler) ShouldSendMessage(destinationClient vms.DestinationClie
 		return false, nil
 	}
 
-	return true, nil
+	var decision bool = true
+	deciderClientValue := reflect.ValueOf(m.deciderClient)
+	if deciderClientValue.IsValid() && !deciderClientValue.IsNil() {
+		warpMsgIDStr := m.unsignedMessage.ID().Hex()
+
+		warpMsgID, err := hex.DecodeString(warpMsgIDStr)
+		if err != nil {
+			m.logger.Error(
+				"Error decoding message ID",
+				zap.String("warpMsgIDStr", warpMsgIDStr),
+				zap.Error(err),
+			)
+			return decision, err
+		}
+
+		// TODO: add a timeout to the context
```

Comment:
Let's include timeouts in this PR

Reply:
addressed in https://github.com/ava-labs/awm-relayer/pull/344/commits/f89bd35a55efdf0a470a13d42f44de78921490b1